### PR TITLE
Ability to run an external binary instead of hyperkube cloud-controller-manager

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

Since we want folks to test their own binaries, let's allow
a way for them them to specify a custom binary.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -646,8 +647,14 @@ function start_cloud_controller_manager {
       node_cidr_args="--allocate-node-cidrs=true --cluster-cidr=10.1.0.0/16 "
     fi
 
+    if [[ -z ${EXTERNAL_CLOUD_PROVIDER_EXE} ]]; then
```
Comment (sttts) [inline hack/local-up-cluster.sh]: EXE is so DOS'ish. Don't we call that "binary" or "command"?

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -646,8 +647,14 @@ function start_cloud_controller_manager {
...
+      CLOUD_CTLRMGR="${CONTROLPLANE_SUDO} ${GO_OUT}/hyperkube cloud-controller-manager"
+    else
+      CLOUD_CTLRMGR="${CONTROLPLANE_SUDO} ${EXTERNAL_CLOUD_PROVIDER_EXE}"
+    fi
+
     CLOUD_CTLRMGR_LOG=${LOG_DIR}/cloud-controller-manager.log
-    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" cloud-controller-manager \
+    ${CLOUD_CTLRMGR} \
```
Comment (sttts) [inline hack/local-up-cluster.sh]: `${EXTERNAL_CLOUD_PROVIDER_EXE:-${GO_OUT}/hyperkube cloud-controller-manager}`

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (dims) [reply hack/local-up-cluster.sh]: :) ack switching to BINARY

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (dims) [reply hack/local-up-cluster.sh]: y much better. fixing

**Comment by deads2k** [comment]
why do we use hyperkube for this at all?

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -647,7 +648,7 @@ function start_cloud_controller_manager {
     fi
 
     CLOUD_CTLRMGR_LOG=${LOG_DIR}/cloud-controller-manager.log
-    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" cloud-controller-manager \
+    ${EXTERNAL_CLOUD_PROVIDER_BINARY:-${GO_OUT}/hyperkube cloud-controller-manager} \
```
Comment (sttts) [inline hack/local-up-cluster.sh]: We lost sudo. Am happy if it still works. Every sudo that is gone is a good sudo.

Comment (dims) [reply hack/local-up-cluster.sh]: hmmm, we'll need it for openstack to get to config drive. will add it back in a few

Comment (sttts) [reply hack/local-up-cluster.sh]: concerning that you need root access for a controller, isn't it?

Comment (dims) [reply hack/local-up-cluster.sh]: yep. will have to figure out a way to remove that dependency (for config drives)

File: hack/local-up-cluster.sh [hunk 4]
```diff
@@ -647,7 +648,7 @@ function start_cloud_controller_manager {
     fi
 
     CLOUD_CTLRMGR_LOG=${LOG_DIR}/cloud-controller-manager.log
-    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" cloud-controller-manager \
+    ${CONTROLPLANE_SUDO} ${EXTERNAL_CLOUD_PROVIDER_BINARY:-${GO_OUT}/hyperkube cloud-controller-manager} \
```
Comment (lavalamp) [inline hack/local-up-cluster.sh]: Please fix the quoting. Spaces in paths are evil but shouldn't cause failures.

**Comment by dims** [comment]
@sttts @deads2k this is ready. PTAL.

**Pull request merged**